crowbar_framework/lib/tasks/assets.rake

```diff
@@ -0,0 +1,74 @@
+#
+# Copyright 2015, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+namespace :assets do
+  def available_assets
+    Pathname.glob(
+      Rails.root.join(
+        "public",
+        "assets",
+        "**",
+        "*"
+      )
+    )
+  end
+
+  def digested_regex
+    /(-{1}[a-z0-9]{32}*\.{1}){1}/
+  end
+
+  task setup_logger: :environment do
+    @logger = Logger.new(STDOUT)
+    @logger.level = Logger::INFO
+  end
+
+  task non_digested: :setup_logger do
+    available_assets.each do |asset|
+      next if asset.directory?
+      next unless asset.to_s =~ digested_regex
+
+      simple = asset.dirname.join(
+        asset.basename.to_s.gsub(digested_regex, ".")
+      )
+
+      if simple.exist?
+        simple.delete
+      end
+
+      @logger.info "Symlinking #{simple}"
+      simple.make_symlink(asset.basename)
+    end
+  end
+
+  task clean_dangling: :setup_logger do
+    available_assets.each do |asset|
+      next if asset.directory?
+      next if asset.to_s =~ digested_regex
+
+      next unless asset.symlink?
+
+      unless asset.realpath.exist?
```

Comment:
```realdirpath``` and ```readlink``` are missing the sub folders for files in subfolders. For example ```/projects/cloud/cloud-5/barclamps/crowbar/crowbar_framework/public/assets/bootstrap/glyphicons-halflings-regular.woff``` gets to ```/projects/cloud/cloud-5/barclamps/crowbar/crowbar_framework/glyphicons-halflings-regular-409673797ad7e6ab108d5e8eb5dac800.woff```.